packages/sources/view-function-multi-chain/src/transport/function.ts

```diff
@@ -52,25 +54,78 @@ export class MultiChainFunctionTransport extends SubscriptionTransport<MultiChai
     await this.responseCache.write(this.name, [{ params: param, response }])
   }
 
-  async _handleRequest(
+  async _handleRequestMultiChain(
     param: RequestParams,
   ): Promise<AdapterResponse<MultiChainFunctionTransportTypes['Response']>> {
-    const { address, signature, inputParams, network } = param
+    const { network } = param
 
     const networkName = network.toUpperCase()
     const networkEnvName = `${networkName}_RPC_URL`
     const chainIdEnvName = `${networkName}_CHAIN_ID`
 
     const rpcUrl = process.env[networkEnvName]
-    const chainId = Number(process.env[chainIdEnvName])
 
-    if (!rpcUrl || isNaN(chainId)) {
+    // The Starknet ChainIds are too large to fit into the JS Number type
+    const chainId = new BigNumber(process.env[chainIdEnvName] ?? NaN)
+
+    if (!rpcUrl || chainId.isNaN()) {
       throw new AdapterInputError({
         statusCode: 400,
-        message: `Missing '${networkEnvName}' or '${chainIdEnvName}' environment variables.`,
+        message: `Missing '${networkEnvName}': '${rpcUrl}' or '${chainIdEnvName}': '${chainId.toString()}' environment variables.`,
       })
     }
 
+    if (
+      !chainId.equals(constants.StarknetChainId.SN_SEPOLIA) &&
+      !chainId.equals(constants.StarknetChainId.SN_MAIN)
+    ) {
```

Comment:
Can we use `networkName` rather than `chainId` for determining the network?

Reasoning is that `networkName` comes from the input params, and so are guaranteed to be consistent across nodes running the same job spec. `chainId` comes from the EA config, and so may differ depending on how the deployer has configured the EA. 

Happy path should have the same effect, but using `networkName` should eliminate a class of tough to debug issues.

Reply:
We can use the network name yes, and just have two special cases for 'STARKNET_SEPOLIA' and 'STARKNET_MAINNET'.

Reply:
Apologies for the churn but I actually take back the above - if we rely on `networkName` then the EA has to know what network name will be used by the job spec & deployed EA, which don't want to hardcode.

Reply:
Throwing this out there ~ can't we have a new input parameter like `"chainType": "evm"` or `"chainType": "starknet"`? Would help us test more than one

Reply:
If we go with the chain id approach can we use the ASCII string representation? So `SN_SEPOLIA` and `SN_MAIN` vs the hexadecimal. I think the hexadecimal, for a node operator is more error prone. We can convert it to the hexadecimal representation in the code here

Reply:
(the starknet chainId is a hexadecimal representation of the network name)